src/components/TradeWidget/TxMessage.tsx

```diff
@@ -1,21 +1,98 @@
-import React from 'react'
+import React, { useMemo } from 'react'
 import { formatTimeInHours } from 'utils'
 import { useFormContext } from 'react-hook-form'
 import { TokenDetails } from 'types'
 import styled from 'styled-components'
 import { TradeFormData } from '.'
 import { displayTokenSymbolOrLink } from 'utils/display'
+import { usePriceEstimationInOwl } from 'hooks/usePriceEstimation'
+import BigNumber from 'bignumber.js'
+import { ZERO_BIG_NUMBER } from 'const'
+
+import alertIcon from 'assets/img/alert.svg'
+import { useGasPrice } from 'hooks/useGasPrice'
 
 interface TxMessageProps {
   sellToken: TokenDetails
   receiveToken: TokenDetails
+  networkId: number
 }
 
 const TxMessageWrapper = styled.div`
   padding: 1em;
 `
 
-export const TxMessage: React.FC<TxMessageProps> = ({ sellToken, receiveToken }) => {
+const Warning = styled.p`
+  background: beige;
+  padding: 0.5em;
+  box-shadow: inset 0 0 7px 3px #cc262647;
+  border-radius: 0.3em;
+  display: flex;
+
+  .alert {
+    min-width: 2.5em;
+  }
+`
+
+const DEFAULT_GASP_RICE = 40e9 // 40 Gwei
+const ETH_PRICE_IN_OWL = 240 * 1000
+const SUBSIDIZE_FACTOR = 10
+
+const SETTLEMENT_FACTOR = 1.5
+const FEE_FACTOR = 1000
+
+const calcMinTradableAmountInOwl = (gasPrice: number): BigNumber => {
+  //                            trade tx gasLimit
+  const MIN_ECONOMICAL_VIABLE_FEE_IN_OWL = 120000 * gasPrice * ETH_PRICE_IN_OWL
+
+  const MIN_FEE = MIN_ECONOMICAL_VIABLE_FEE_IN_OWL / SUBSIDIZE_FACTOR
```

Comment:
I would also add a buffer. Like multiply 

`MIN_FEE = BUFFER_TO_ACCOUNT_FOR_TOKEN_AND_GAS_PRICE_CHANGES  * MIN_ECONOMICAL_VIABLE_FEE_IN_OWL / SUBSIDIZE_FACTOR`

i.e.
`BUFFER_TO_ACCOUNT_FOR_TOKEN_AND_GAS_PRICE_CHANGES  = 1.20`   ---> account for a 20% change in the time it takes to mine the tx, and start the batch

Of course with a shorter name, I just made it loooong for illustration purposes